frontend/src/pages/connectionTypes/CreateConnectionType/CreateConnectionTypePage.tsx

```diff
@@ -0,0 +1,126 @@
+
+type CreateConnectionTypePageProps = {
+  prefillNameDesc?: NameDescType;
+  prefillEnabled?: boolean;
+  prefillFields?: ConnectionTypeField[];
+};
+
+export const CreateConnectionTypePage: React.FC<CreateConnectionTypePageProps> = ({
+  prefillNameDesc,
+  prefillEnabled,
+  prefillFields,
+}) => {
+  const [isDrawerExpanded, setIsDrawerExpanded] = React.useState(false);
+
+  const [connectionNameDesc, setConnectionNameDesc] = React.useState<NameDescType>(
+    prefillNameDesc || {
+      name: '',
+      k8sName: undefined,
+      description: '',
+    },
+  );
+  const [connectionEnabled, setConnectionEnabled] = React.useState<boolean>(
+    prefillEnabled || false,
+  );
+  const [connectionFields] = React.useState<ConnectionTypeField[]>(prefillFields || []);
+
+  return (
+    <ConnectionTypePreviewDrawer
+      isExpanded={isDrawerExpanded}
+      onClose={() => setIsDrawerExpanded(false)}
+      obj={createConnectionTypeObj(
+        {
+          k8sName: translateDisplayNameForK8s(connectionNameDesc.name),
+          displayName: connectionNameDesc.name,
+          description: connectionNameDesc.description,
+          enabled: connectionEnabled,
+          username: '',
+        },
+        connectionFields,
+      )}
+    >
+      <ApplicationsPage
+        title="Create connection type"
+        loaded
+        empty={false}
+        errorMessage="Unable load to connection types"
+        breadcrumb={<CreateConnectionTypeBreadcrumbs />}
+        headerAction={
+          <Button
+            variant="secondary"
+            icon={<OpenDrawerRightIcon />}
+            aria-expanded={isDrawerExpanded}
+            onClick={() => setIsDrawerExpanded(!isDrawerExpanded)}
+            data-testid="preview-drawer-toggle-button"
+          >
+            Preview
+          </Button>
+        }
+      >
+        <PageSection isFilled variant="light">
+          <Form>
+            <FormSection title="Type details" style={{ maxWidth: 625 }}>
+              <NameDescriptionField
+                nameFieldId="connection-type-name"
+                nameFieldLabel="Connection type name"
+                descriptionFieldId="connection-type-description"
+                descriptionFieldLabel="Connection type description"
+                data={connectionNameDesc}
+                setData={setConnectionNameDesc}
+                autoFocusName
+              />
+              <FormGroup label="Enable">
+                <Checkbox
+                  label="Connection is enabled and therefore available to use by users in your org"
+                  id="connection-type-enable"
+                  name="connection-type-enable"
+                  data-testid="connection-type-enable"
+                  isChecked={connectionEnabled}
+                  onChange={(_e, value) => setConnectionEnabled(value)}
+                />
+              </FormGroup>
+            </FormSection>
+            <FormSection title="Fields">
```

Comment:
Would be better to remove the top margin
```suggestion
            <FormSection className="pf-v5-u-mt-0" title="Fields">
```

Reply:
updated (update images are in the pr description)